### PR TITLE
feat: allow masked otp input

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
Adds support for masked input fields in `OTPInput`

**Demo:** 

https://user-images.githubusercontent.com/24487274/204982587-e1400d43-7f01-4bc5-8b85-6072cdb473a9.mov

### CODE REVIEW

File: packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap [hunk 1]
```diff
@@ -332,7 +332,7 @@ exports[`<OTPInput /> should render 1`] = `
                   inputmode="decimal"
                   placeholder=""
                   required=""
-                  type="text"
+                  type="numeric"
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: Numeric?

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: Keyboard type is decimal, wanted to make the type similar too. Should've done thing before only

**Review by anuraghazra — APPROVED** [review]
Mostly LGTM

File: packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: Related to this right? https://github.com/razorpay/blade/issues/786 
Hopefully no one is depending on OTP being alphanumeric.

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: Probably not. We can revisit this if we see someone facing any issues. Don't think we need to work on this right now

File: packages/blade/src/components/Input/OTPInput/OTPInput.tsx [hunk 2]
```diff
@@ -243,6 +248,7 @@ const OTPInput = ({
             errorText={errorText}
             helpText={helpText}
             hideFormHint={true}
+            type={isMasked ? 'password' : 'numeric'}
```
Comment (saurabhdaware) [inline packages/blade/src/components/Input/OTPInput/OTPInput.tsx:251]: Shouldn't this be `type="text"` and `inputmode="numeric"`? couldn't find any mdn documentation of `numeric` type as such https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#using_number_inputs

File: packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: So `numeric` won't really stop that. We might have to add `pattern="\d*"` for this.
image

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: Yeah, it won't stop it. Just made it consistent with the keyboardType. Whether we want to stop it or not can be tackled separately. Lets scope this PR to just addition of masked input. I added `numeric` to make it consistent with they keyboardType but I can revert it if its an issue 😅

Comment (saurabhdaware) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: Oh yes yes makes sense. Only this concern then https://github.com/razorpay/blade/pull/897#discussion_r1036766905. Rest looks good to me.

File: packages/blade/src/components/Input/OTPInput/OTPInput.tsx (hunk 2 above)
Comment (kamleshchandnani) [reply packages/blade/src/components/Input/OTPInput/OTPInput.tsx:251]: @chaitanyadeorukhkar - https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Input/BaseInput/_decisions/_decisions.md#why-we-are-not-having-keyboardtypenumeric

File: packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Input/OTPInput/__tests__/__snapshots__/OTPInput.web.test.tsx.snap:335]: AHH makes sense. So numeric will change the keyboard behaviour not the input's validation.

File: packages/blade/src/components/Input/OTPInput/OTPInput.tsx (hunk 2 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/OTPInput.tsx:251]: Yeah, seems like type `numeric` doesn't exist. I used `BaseInput`'s props only. We have specific `numeric` as an acceptable value for type. Let me remove this change. For now I'll just keep it as is and raise an issue for it.

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/OTPInput.tsx:251]: @kamleshchandnani the link you specified is for `keyboardType` attribute not for `type`. `type=numeric` doesn't exist but we accept that value. We should ideally make it `type=number`?

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/OTPInput.tsx:251]: Went through [this](https://css-tricks.com/better-form-inputs-for-better-mobile-user-experiences/)

So we don't want to pass `type=number` instead we want to pass `inputmode=decimal`. inputmode is `keyboardType` for us which we're already passing. `type=numeric` seems to be an incorrect value we're accepting and rather it should be `type=number`. I'll make that change in a different PR

Screenshot 2022-12-01 at 4 34 34 PM

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/OTPInput.tsx]: Raised a PR for that change

**Pull request merged**